code/modules/mob/living/carbon/human/species.dm

```diff
@@ -731,6 +729,10 @@
 	return
 
 /datum/species/proc/handle_chemicals(datum/reagent/chem, mob/living/carbon/human/H)
+	if(chem.id == exotic_blood)
+		H.blood_volume = min(H.blood_volume + round(chem.volume, 0.1), BLOOD_VOLUME_MAXIMUM)
+		H.reagents.remove_reagent(chem.id)
+		return 1
```

Comment:
if a chemical in our reagents match our blood substance, we absorb it to add it to our blood volume.